ocaml/parsing/parsetree.mli

```diff
@@ -47,6 +47,8 @@ type location_stack = Location.t list
 
 type modality = | Modality of string [@@unboxed]
 
+type mode = | Mode of string [@@unboxed]
+
```

Comment:
It seems that `mode loc list` is referred to a lot, which suggests that maybe we should make a dedicated type (as it was in the previous encoding). In fact, maybe `mode` should be defined as
```
type mode = | Mode of string loc
```
because `type mode` is the result of parsing, and thus natural to include a `loc`. This is similar to `type expression`, `type attribute`, etc.

And then
```
type modes = mode list loc
```
I'm less opioniated about the `loc` here - but still think it's better included.

Reply:
(As discussed, we think this change is too big that would involve a lot in the tree)